assets/src/apps/page-editor/PageEditor.tsx

```diff
@@ -338,9 +339,17 @@ export class PageEditor extends React.Component<PageEditorProps, PageEditorState
 
     if (item !== undefined) {
       if (item.undoable.type === 'PageUndoable') {
-        this.update({
-          content: this.state.content.insertAt(item.undoable.index, item.undoable.item),
-        });
+        if (this.state.content.find(item.contentKey)) {
```

Comment:
In order to support pushing an undoable of a nested item, this was modified to check if the undoable being restored exists with a certain unique key. If it does, then the item is replaced with the undoable (restore a nested change/deletion). If not, then the item is inserted (restore a top level deletion)